includes/senders/BaseSender.php

```diff
@@ -36,16 +44,26 @@ public static function json_unserialize( $json ) {
 
 		$data = json_decode( $json, true );
 
-		$name = ! empty( $data['name'] ) ? $data['name'] : '';
+		$name  = ! empty( $data['name'] ) ? $data['name'] : '';
+		$image = ! empty( $data['image'] ) ? WPNotify_BaseImage::json_unserialize( json_encode( $data['image'] ) ) : null;
```

Comment:
What happens if we set the `$associative` parameter of `json_decode()` to false? Do we have our properties correctly instantiated as object? 

If this doesn't work, could we use `$depth` parameter of the `json_decode()` function to only decode the first level objects? We may able able to have our image as a JSON string and then not have to encode it again.

Reply:
I've changed the way the image is initialised. I've used the same logic from `WPNotify_BaseNotification` instantiating the image from reflection.